Source/WebCore/css/CSSSelector.cpp

```diff
@@ -864,4 +864,24 @@ bool CSSSelector::hasExplicitPseudoClassScope() const
     return visitAllSimpleSelectors(check);
 }
 
+bool CSSSelector::isHostPseudoClass() const
+{
+    return match() == CSSSelector::Match::PseudoClass && pseudoClass() == CSSSelector::PseudoClass::Host;
+}
+
+bool CSSSelector::isOrContainsHostPseudoClass() const
+{
+    if (isHostPseudoClass())
+        return true;
+
+    if (const auto* selectorList = this->selectorList()) {
+        for (const auto& subselector : *selectorList) {
+            if (subselector.isOrContainsHostPseudoClass())
+                return true;
+        }
+    }
+
+    return false;
+}
```

Comment:
Are this and the call sites correct for `:not(:host)` and all other pseudo classes taking selector list arguments? Should this be limited to `:is()` and `:where()`?